File: src/main/java/com/booknav/springbackend/dto/Book/BookDTO.java
package com.booknav.springbackend.dto.Book;

import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
public class BookDTO {
    private Integer bookId;
    @NotBlank(message = "Book title is required")
    @Size(max = 150, message = "Book title can't exceed 150 characters.")
    private String title;
    @NotBlank(message = "Book description is required")
    @Size(max = 300, message = "Book description can't exceed 300 characters.")
    private String description;
    @NotBlank(message = "Book publish year is required")
    @Pattern(regexp = "\\b(1[0-9]{3}|[2-9][0-9]{3})\\b", message = "Please enter a valid year")
    private Integer publishYear;
    @NotBlank(message = "Book cover's image link is required")
    private String cover;
    @NotBlank(message = "Book price is required")
    @Min(value = 1, message = "Book price can't be less than 1")
    @Max(value = 3000, message = "Book price can't exceed 3000")
    private BigDecimal price;
    private String bookAuthor;
    private BookCatReqDTO bookCategory;
}

File: ASP Core - Backend/Dtos/BookDTO.cs
using System.ComponentModel.DataAnnotations;

namespace aspcorebackend.Dtos {
    public class BookDTO {
        public int BookId { get; set; }

        [Required(ErrorMessage = "Book title is required")]
        [MaxLength(150, ErrorMessage = "Book title can't exceed 150 characters.")]
        public string Title { get; set; } = null!;

        [Required(ErrorMessage = "Book description is required")]
        [MaxLength(300, ErrorMessage = "Book description can't exceed 300 characters.")]
        public string Description { get; set; } = null!;

        [Required(ErrorMessage = "Book publish year is required")]
        [RegularExpression(@"\b(1[0-9]{3}|[2-9][0-9]{3})\b", ErrorMessage = "Please enter a valid year")]
        public int PublishYear { get; set; }

        [Required(ErrorMessage = "Book cover's image link is required")]
        public string Cover { get; set; } = null!;

        [Required(ErrorMessage = "Book price is required")]
        [Range(1, 3000, ErrorMessage = "Book price must be between 1 and 3000")]
        public decimal Price { get; set; }

        public string? BookAuthor { get; set; }

        public string? BookCategory { get; set; }
    }
}
